src/core/model/types/numeric_type.h

```diff
@@ -31,6 +38,28 @@ class INumericType : public IMetrizableType {
     [[nodiscard]] virtual std::byte const* Max() const = 0;
 };
 
+template <typename T>
+T INumericType::GetValueAs(std::byte const* value) const {
+    switch (GetTypeId()) {
+        case TypeId::kDouble: {
+            model::Double data = INumericType::GetValue<Double>(value);
+            return static_cast<T>(data);
+        }
+        case TypeId::kInt: {
+            model::Int data = INumericType::GetValue<Int>(value);
+            return static_cast<T>(data);
+        }
+        default: {
+            assert(false);
+            __builtin_unreachable();
+        }
+    }
+}
+
+inline void INumericType::CastTo(std::byte* value, INumericType const& to) const {
+    CastTo(value, to.GetTypeId());
+}
```

Comment:
This function is small enough to keep it's definition inside the class. Initially I asked to move the definition out of the class because the definition was big enough to made it harder to read the class interface.